src/tests/test_log_client.py

```diff
@@ -0,0 +1,226 @@
+import unittest
+import os
+import pwd
+import StringIO
+import threading
+import mock
+import e_errors
+import enstore_functions
+import log_client
+import udp_client
+import mock_csc
+""" Test the log_client module
+    The log_client module is used to send log messages to the log server.
+    Author: Dennis Box
+    Date: 2023-07-10
+"""
+
+
+class TestLoggerClient(unittest.TestCase):
+
+    def setUp(self):
+        self.sent_msg = mock.MagicMock()
+        self.user = pwd.getpwuid(os.getuid())[0]
+        udp_client.UDPClient.send = self.sent_msg
+        udp_client.UDPClient.send_no_wait = self.sent_msg
+        csc = mock_csc.csc()
+        self.log_client = log_client.LoggerClient(csc)
+
+    def test___init__(self):
+        self.assertTrue(isinstance(self.log_client, log_client.LoggerClient))
+
+    def test_log_func(self):
+        self.sent_msg.reset_mock()
+        t = 0.0
+        pid = 9999999
+        name = self.log_client.name
+        args = [e_errors.MISC + 1, "an log message", "info"]
+        self.log_client.log_func(t, pid, name, args)
+        unique_id = True
+        p1 = {
+            'message': '9999999 %s M LOG_CLIENT  6 an log message' % self.user,
+            'work': 'log_message'}
+        p2 = ('131.225.214.78', 7504)
+        self.sent_msg.assert_called_with(p1, p2, unique_id=True)
+
+    def test_set_get_logpriority(self):
+        # NB the log server reacts to the log priority
+        # this needs to be tested with the log server running
+        # or in the log server test
+        log_prio = 99999
+        self.assertNotEqual(self.log_client.get_logpriority(), log_prio)
+        self.log_client.set_logpriority(log_prio)
+        self.assertEqual(self.log_client.get_logpriority(), log_prio)
+
+    def test_get_logfile_name(self):
```

Comment:
I feel like these 3 could (maybe should?) be parameterized with
{
f: self.log_client.get_logfile_name, m: {'work': 'get_logfile_name'},
f: functools.partial(self.log_client.get_logfiles, 'today'), m: {'work': 'get_logfiles', 'period': 'today'},
f: self.log_client.get_last_logfile_name, m: {'work': 'get_last_logfile_name'},
}
but.. that might be a notable decrease in readability. Up to you!

Reply:
I have elected to keep this more readable

Reply:
Ack, sounds good